lib/mirage.ml

```diff
@@ -1489,13 +1489,15 @@ let configure_makefile ~target ~root ~name ~warn_error info =
     (match target with
       `Unix | `MacOSX -> "unix" | `Xen -> "xen" | `Virtio | `Ukvm -> "solo5" );
   append fmt "SYNTAX += -tag-line \"<static*.*>: warn(-32-34)\"\n";
-  append fmt "LD?=ld";
   append fmt "BUILD  = ocamlbuild -use-ocamlfind -tags $(TAGS) $(LIBS) $(SYNTAX) $(FLAGS)\n\
               OPAM   = opam\n\n\
               export PKG_CONFIG_PATH=$(shell opam config var prefix)\
               /lib/pkgconfig\n\n\
               export OPAMVERBOSE=1\n\
               export OPAMYES=1";
+  (match target with
+   | `Virtio -> append fmt "LD=$(shell pkg-config solo5-kernel-virtio --variable=ld)"
```

Comment:
1. I'd make this case apply to both `Virtio` and `Ukvm` _targets_ for consistency -- solo5/solo5#102 deliberately introduces the `LD` variable for both even though it is only used by `Virtio`.
2. Can you use `LD?=$(shell ...)` instead of `LD=`, i.e. allow user override?